tests/integration/test.replication.js

```diff
@@ -217,6 +232,9 @@ adapters.forEach(function (adapters) {
         result.docs_written.should.equal(0, 'correct # docs written (1)');
         return db.info();
       }).then(function (info) {
+        if (testUtils.isSyncGateway()) {
+          return [{"ok": {"_id": "doc", "_rev": doc1._rev}}];
+        }
```

Comment:
This patch is about running as many assertions as possible without actually touching implementation unless I have to. So my goal is to build a process around cutting down the number of `isSyncGateway` calls over time. I agree this is an important one to work on